pkg/mod/fs.go

```diff
@@ -24,19 +24,23 @@ func (fs *Fs) Open(name string) (afero.File, error) {
 		return f, nil
 	}
 
-	// filepath.Join will strip path elements of ".", so if the root is "."
-	// it will still work as a go module path when prepended with "."
-	name = filepath.Join(fs.root, name)
-	mod, err := Find(name)
-	if err != nil {
-		return nil, err
-	}
-	relpath, err := filepath.Rel(mod.Name, name)
-	if err != nil {
-		return nil, err
+	if SyslModules {
+		// filepath.Join will strip path elements of ".", so if the root is "."
+		// it will still work as a go module path when prepended with "."
+		name = filepath.Join(fs.root, name)
+		mod, err := Find(name)
+		if err != nil {
+			return nil, err
+		}
+		relpath, err := filepath.Rel(mod.Name, name)
+		if err != nil {
+			return nil, err
+		}
+
+		return syslutil.NewChrootFs(afero.NewOsFs(), mod.Dir).Open(relpath)
 	}
 
-	return syslutil.NewChrootFs(afero.NewOsFs(), mod.Dir).Open(relpath)
+	return nil, err
 }
 
 func (fs *Fs) Create(name string) (afero.File, error) {
```

Comment:
Something I noticed when I looked at this code before...

If you embed `afero.Fs` into `mod.Fs`, you can get rid of all these functions that just delegate to the underlying `afero.Fs`:

```Go
type Fs struct {
        afero.Fs
	root   string
}

func NewFs(fs afero.Fs, root string) *Fs {
	return &Fs{Fs: syslutil.NewChrootFs(fs, root), root: root}
}
```

Then all the methods of `afero.Fs` will be promoted into `mod.Fs` and you don't need to implement the delegation manually.

In `Open()`, you would use `fs.Fs.Open(...)` instead of `fs.source.Open(...)`.

This would also increase test coverage because none of the delegations are tested.

Reply:
Thanks cam! This (`fs.Fs`) is what I was looking for.

Reply:
From https://golang.org/ref/spec#Struct_types:

> The unqualified type name acts as the field name.